#FormationMessage formation
#VectorMessage frp
#Relationship[] relationships
#VectorMessage linear_error
float64 angular_error
int32 timestep
int32 reference_id
float64 temperature
float64 heat